### PR TITLE
Add persistent volume claim retention policy

### PR DESCRIPTION
**Pull request by nirroz93**
---
Add persistentVolumeClaimRetentionPolicy available since k8s 1.23 (alpha) and 1.27 (beta). With logs, this makes a lot of sense (with remote logging only running task log is important, and on scale down pvc should be removed)

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -617,6 +617,11 @@ workers:
   persistence:
     # Enable persistent volumes
     enabled: true
+    # persistentVolumeClaimRetentionPolicy if statefulset is used, to determine delete when scaled\removed
```
Comment (amoghrajesh) [inline chart/values.yaml:620]: nit:
```suggestion
    # persistentVolumeClaimRetentionPolicy if statefulset is used, to determine delete when scaled/removed
```

File: chart/values.yaml [hunk 2]
```diff
@@ -1426,6 +1431,8 @@ triggerer:
...
```
Comment (amoghrajesh) [inline chart/values.yaml:1434]:
```suggestion
    # persistentVolumeClaimRetentionPolicy if statefulset is used, to determine delete when scaled/removed
```

**Review by amoghrajesh — COMMENTED** [review]
@nirroz93 good work on the PR. Few comments:

1. Can we add this also to `chart/files/pod-template-file.kubernetes-helm-yaml` for K8sExecutor?
2. Can we add unit tests foe this under `helm_tests`? Check similar tests.

**Comment by romsharon98** [comment]
> @nirroz93 good work on the PR. Few comments:
> 
> 1. Can we add this also to `chart/files/pod-template-file.kubernetes-helm-yaml` for K8sExecutor?
> 2. Can we add unit tests foe this under `helm_tests`? Check similar tests.



> @nirroz93 good work on the PR. Few comments:
> 
> 1. Can we add this also to `chart/files/pod-template-file.kubernetes-helm-yaml` for K8sExecutor?
> 2. Can we add unit tests foe this under `helm_tests`? Check similar tests.

no need to add this to `chart/files/pod-template-file.kubernetes-helm-yaml` because this feature belongs only to statefulset.

**Comment by amoghrajesh** [comment]
@nirroz93 are you actively working on this?

**Pull request closed without merging**